sdk/avalon_sdk/ethereum/unit_tests/test_ethereum_work_order_impl.py

```diff
@@ -0,0 +1,170 @@
+    def __init__(self, config_file):
+        super(TestEthereumWorkOrderProxyImpl, self).__init__()
+        if not path.isfile(config_file):
+            raise FileNotFoundError("File not found at path: {0}".format(
+                path.realpath(config_file)))
+        try:
+            with open(config_file) as fd:
+                self.__config = toml.load(fd)
+        except IOError as e:
+            if e.errno != errno.ENOENT:
+                raise Exception("Could not open config file: %s", e)
+        self.__eth_conn = EthereumWorkOrderProxyImpl(self.__config)
+
+    def test_work_order_submit_positive(self):
+        read_json = read_json_file(
+            "work_order_req.json", ["./"])
+        wo_id = urandom(32).hex()
+        worker_id = urandom(32).hex()
+        requester_id = urandom(32).hex()
+
+        wo_req = json.loads(read_json)["params"]
+        wo_req["workOrderId"] = wo_id
+        wo_req["workerId"] = worker_id
+        wo_req["requesterId"] = requester_id
+
+        result = self.__eth_conn.work_order_submit(wo_id,
+                                                   worker_id,
+                                                   requester_id,
+                                                   json.dumps(wo_req))
+        self.assertEqual(result, 0, "Work order should pass")
+
+    def test_work_order_submit_mismatch(self):
+        read_json = read_json_file(
+            "work_order_req.json", ["./"])
+        wo_id = urandom(32).hex()
+        worker_id = urandom(32).hex()
+        requester_id = urandom(32).hex()
+
+        wo_req = json.loads(read_json)["params"]
+        wo_req["workOrderId"] = wo_id
+        wo_req["workerId"] = urandom(32).hex()  # Different workerId
+        wo_req["requesterId"] = requester_id
+
+        result = self.__eth_conn.work_order_submit(wo_id,
+                                                   worker_id,
+                                                   requester_id,
+                                                   json.dumps(wo_req))
+        self.assertEqual(result, 1, "Work order sumbissoin should fail")
+
+    def test_work_order_get_result(self):
+        pass
+
+    def test_work_order_complete(self):
+        read_json = read_json_file(
+            "work_order_get_result.json", ["./"])
+        wo_id = urandom(32).hex()
+
+        result = self.__eth_conn.work_order_complete(wo_id,
+                                                     read_json)
+        self.assertEqual(
+            result, 0, "Work order result submission should succeed")
+
+    def test_work_order_complete_error(self):
+        read_json = read_json_file(
+            "work_order_get_result_error.json", ["./"])
+        wo_id = urandom(32).hex()
+
+        result = self.__eth_conn.work_order_complete(wo_id,
+                                                     read_json)
+        self.assertEqual(
+            result, 0, "Work order result submission should succeed")
```

Comment:
Is the assert message correct?

Reply:
Yes @Ram-srini . The work_order_complete function should successfully write the result(execution result or error in execution) to the blockchain.